include/pinocchio/algorithm/model.hxx

```diff
@@ -256,12 +273,32 @@ namespace pinocchio
 
     geomModel.geometryObjects.reserve (geomModelA.ngeoms + geomModelB.ngeoms);
 
-    // Copy modelA joints until frame.parentJoint
     details::appendUniverseToModel (modelA, geomModelA, 0, id, model, geomModel);
+    // Compute joints of A that should be added before and after joints of B
+    std::vector<JointIndex> AJointsBeforeB;
+    std::vector<JointIndex> AJointsAfterB;
+    // All joints until the parent of frameInModelA come first
     for (JointIndex jid = 1; jid <= frame.parent; ++jid)
+    {
+      AJointsBeforeB.push_back(jid);
+    }
+    // descendants of the parent of frameInModelA come also before model B
+    for (JointIndex jid = frame.parent+1; jid < modelA.joints.size(); ++jid)
+    {
+      if (hasAncestor(modelA, jid, frame.parent))
+      {
+        AJointsBeforeB.push_back(jid);
+      } else
+      {
+        AJointsAfterB.push_back(jid);
+      }
+    }
```

Comment:
Given the compact organization of the list of joints, I would expect that you can copy all the joints of model A until the first joint whose id `id_insert` matches:
- `id_frame_parent < id_insert`: the joint is after the `frame.parent` where modelB is appended.
- `modelA.parents[id_insert] < id_frame_parent`: the joint parent is before `frame.parent`.

`id_insert` is to be added after model B.

If that is true, the code would *arguably* be more atomic.

Reply:
Note that I'm not specifically asking you to change. I'm done with the current implementation.